api/geth_backend.go

```diff
@@ -1713,24 +1713,31 @@ func (b *GethStatusBackend) prepareForKeycard(request *requests.CreateAccount, i
 		return response, nil
 	}
 
-	kp := wallet.NewKeycardPairings()
-	kp.SetKeycardPairingsFile(response.nodeConfig.KeycardPairingDataFile)
-	pairings, err := kp.GetPairings()
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to get keycard pairings")
-	}
+	if request.KeycardPairingKey == "" {
+		kp := wallet.NewKeycardPairings()
+		kp.SetKeycardPairingsFile(response.nodeConfig.KeycardPairingDataFile)
+		pairings, err := kp.GetPairings()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get keycard pairings")
+		}
 
-	keycard, ok := pairings[request.KeycardInstanceUID]
-	if !ok {
-		return nil, errors.New("keycard not found in pairings file")
+		keycard, ok := pairings[request.KeycardInstanceUID]
+		if !ok {
+			return nil, errors.New("keycard not found in pairings file")
+		}
+
+		response.settings.KeycardPairing = keycard.Key
+		response.account.KeycardPairing = keycard.Key
+	} else {
+		response.settings.KeycardPairing = request.KeycardPairingKey
+		response.account.KeycardPairing = request.KeycardPairingKey
```

Comment:
Would be nice to invert this condition to have the shorter branch to go first.

```suggestion
	if request.KeycardPairingKey != "" {
		response.settings.KeycardPairing = request.KeycardPairingKey
		response.account.KeycardPairing = request.KeycardPairingKey
	} else {
		kp := wallet.NewKeycardPairings()
		kp.SetKeycardPairingsFile(response.nodeConfig.KeycardPairingDataFile)
		pairings, err := kp.GetPairings()
		if err != nil {
			return nil, errors.Wrap(err, "failed to get keycard pairings")
		}

		keycard, ok := pairings[request.KeycardInstanceUID]
		if !ok {
			return nil, errors.New("keycard not found in pairings file")
		}

		response.settings.KeycardPairing = keycard.Key
		response.account.KeycardPairing = keycard.Key
	}
```